subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -167,25 +149,31 @@ int aws_fota_parse_update_job_exec_state_rsp(char *update_rsp_document,
 	return ret;
 }
 
-int aws_fota_parse_notify_next_document(char *job_document,
-		u32_t payload_len, char *job_id_buf, char *hostname_buf,
-		char *file_path_buf)
+int aws_fota_parse_job_execution(char *job_document, u32_t payload_len,
+				 char *job_id_buf, char *hostname_buf,
+				 char *file_path_buf, int *version_number)
 {
 	struct notify_next_obj job;
 	struct job_document_obj *job_doc_obj;
 
-	int ret = json_obj_parse(job_document,
-				 payload_len,
+	int ret = json_obj_parse(job_document, payload_len,
 				 notify_next_obj_descr,
-				 ARRAY_SIZE(notify_next_obj_descr),
-				 &job);
-	job_doc_obj = &job.execution.job_document;
+				 ARRAY_SIZE(notify_next_obj_descr), &job);
+
+	if (!(ret & BIT(2))) {
+		/* No execution object in JSON */
+		return 1;
+	}
 
 	/* Check if the execution field of the object has been parsed */
-	if (ret & 0x02) {
+	if (ret & BIT(2)) {
+		job_doc_obj = &job.execution.job_document;
 		if (job.execution.job_id != 0) {
 			strncpy_nullterm(job_id_buf, job.execution.job_id,
-				      AWS_JOBS_JOB_ID_MAX_LEN);
+					AWS_JOBS_JOB_ID_MAX_LEN);
+		}
+		if (job.execution.version_number != 0) {
```

Comment:
This check means the caller is not always guaranteed that `version_number` has been updated when the function returns, and AFAICT there is no way to know (the same goes for the other out-args). Maybe it's better if this function updates the static variables directly?

Reply:
Actually you are correct we should return an error if they fail IMO. As these are field that should exists in the object.

Reply:
Added a check if they are there if not `-ENODATA` is returned.